File: cliphook.java
import java.io.*;
import java.nio.charset.StandardCharsets;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class ClipboardLogger {

    private static final String LOG_FILE = "/home/dein_user/clipboard.log";
    private static final int INTERVAL = 2000; // Millisekunden

    private static String lastHash = "";

    public static void main(String[] args) throws InterruptedException {
        while (true) {
            try {
                String content = getClipboard();
                if (!content.isEmpty()) {
                    String currentHash = sha256(content);
                    if (!currentHash.equals(lastHash)) {
                        logClipboard(content);
                        System.out.println("Neuer Clipboard-Inhalt geloggt: " +
                                (content.length() > 30 ? content.substring(0, 30) + "..." : content));
                        lastHash = currentHash;
                    }
                }
            } catch (Exception e) {
                e.printStackTrace();
            }
            Thread.sleep(INTERVAL);
        }
    }

    private static String getClipboard() throws IOException, InterruptedException {
        Process process = new ProcessBuilder("wl-paste", "-n").start();
        BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
        StringBuilder sb = new StringBuilder();
        String line;
        while ((line = reader.readLine()) != null) {
            sb.append(line).append("\n");
        }
        process.waitFor();
        return sb.toString().trim();
    }

    private static String sha256(String str) throws NoSuchAlgorithmException {
        MessageDigest digest = MessageDigest.getInstance("SHA-256");
        byte[] hash = digest.digest(str.getBytes(StandardCharsets.UTF_8));
        StringBuilder hexString = new StringBuilder();
        for (byte b : hash) {
            String hex = Integer.toHexString(0xff & b);
            if (hex.length() == 1) hexString.append('0');
            hexString.append(hex);
        }
        return hexString.toString();
    }

    private static void logClipboard(String content) {
        try (FileWriter fw = new FileWriter(LOG_FILE, true);
             BufferedWriter bw = new BufferedWriter(fw)) {
            String timestamp = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"));
            bw.write("[" + timestamp + "]\n");
            bw.write(content + "\n\n");
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: cliphook.kt
import java.io.BufferedReader
import java.io.File
import java.io.InputStreamReader
import java.security.MessageDigest
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter
import kotlin.concurrent.thread

const val LOG_FILE = "/home/dein_user/clipboard.log"
const val INTERVAL_MS = 2000L // 2 Sekunden

fun getClipboard(): String {
    return try {
        val process = ProcessBuilder("wl-paste", "-n").start()
        val reader = BufferedReader(InputStreamReader(process.inputStream))
        val content = reader.readText().trim()
        process.waitFor()
        content
    } catch (e: Exception) {
        ""
    }
}

fun sha256(str: String): String {
    val md = MessageDigest.getInstance("SHA-256")
    val hashBytes = md.digest(str.toByteArray(Charsets.UTF_8))
    return hashBytes.joinToString("") { "%02x".format(it) }
}

fun logClipboard(content: String) {
    val timestamp = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"))
    File(LOG_FILE).appendText("[$timestamp]\n$content\n\n")
}

fun main() {
    var lastHash = ""

    while (true) {
        val content = getClipboard()
        if (content.isNotEmpty()) {
            val currentHash = sha256(content)
            if (currentHash != lastHash) {
                logClipboard(content)
                println("Neuer Clipboard-Inhalt geloggt: ${content.take(30)}...")
                lastHash = currentHash
            }
        }
        Thread.sleep(INTERVAL_MS)
    }
}

